packages/styleguide/src/components/Editor/config/elements/link.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  ElementConfigI,
+  ElementFormProps,
+  LinkElement,
+  NormalizeFn,
+} from '../../custom-types'
+import { LinkIcon } from '../../../Icons'
+import React, { useEffect, useRef, useState } from 'react'
+import Field from '../../../Form/Field'
+import { Editor, Transforms } from 'slate'
+import {
+  getFullUrl,
+  getLinkInText,
+  isValidHttpUrl,
+} from '../../Core/helpers/text'
+import renderAsText from '../../Render/text'
+import { useSlate } from 'slate-react'
+import RepoSearch from '../../Forms/RepoSearch'
+
+const Form: React.FC<ElementFormProps<LinkElement>> = ({
+  element,
+  path,
+  onChange,
+}) => {
+  const href = element.href || ''
+  const [text, setText] = useState<string>(renderAsText([element]))
+  const editor = useSlate()
+  const textRef = useRef<string>()
+  textRef.current = text
+
+  // Directly updating the link props causes issues
+  // with the normalizer and the cursor.
+  useEffect(() => {
+    return () => {
+      Transforms.insertText(editor, textRef.current, {
+        at: path,
+      })
+    }
+  }, [])
```

Comment:
This transform breaks any marks used inside the link text, e.g. CO<sub>2</sub> will not survive the link modal unmounting.

I would recommend to remove the feature of editing text inside of the modal. Editing within slate is sufficient IMO.

Reply:
I find it quite useful to be able to see and edit everything in one place. I went for a compromise, for now:

- only apply the transform if the text has been changed
- warning about formatting in the form